### DISCUSSION TITLE
DVR Clipping

### DISCUSSION THREAD

**Discussion by nums** [OvenMediaLabs/OvenMediaEngine#1123] [Ideas] [answered]
It would be nice to be able to set a start and end timestamp in order to be able to create extracts from a live DVR.
Maybe like that : https://my-ovenmedia-server/app/live.m3u8?start=123&end=130
With that kind of url, I get only seven second of the live started at 123 seconds.

**Comment by getroot** [comment]
If I think simply, it may be meaningful in VoD, but it does not seem to be meaningful in Live. (because it will be deleted when the live ends)
Can you introduce some use cases?

**Comment by nums** [comment]
On a live with several speakers, it is possible that we only want to retrieve the intervention of one person:
A small example below:
- from 1 minute to 5 minutes: introduction
- 5 minutes to 7 minutes: the word of the president
- 7 minutes to 35 minutes: report on annual figures.

During the live, we would like to create an extract with only the word of the president (a web extract or an mp4 archive)

**But maybe ffmpeg can already do it with some settings, I'll test and let you know.**

The other subject is indeed the duration of the DVR, I thought to create another discussion but I can talk about it here:
If a person is whatching the Live DVR with 1 hour late and the live is over, the DVR cache is deleted and the user can't watch the 1 hour end. Would it be possible to configure a DVR cache lifetime?

**Reply by getroot** [reply, marked as answer]
To be precise, OvenMediaEngine calls what you call a DVR a Live Rewind. Its lifetime is until the live broadcast ends.
https://airensoft.gitbook.io/ovenmediaengine/streaming/low-latency-hls#live-rewind

Perhaps the feature you want is HLS Dump. This creates a permanent HLS VoD. With a file-based HTTP server like nginx you can do this as a VoD service.
https://airensoft.gitbook.io/ovenmediaengine/streaming/low-latency-hls#dump

In the case of Youtube, after the live broadcast ends, it is converted to VoD. If you use the above two functions, you will be able to service similarly.

**Reply by nums** [reply]
Thank you for the clarification.
I just tested the HLS dump and it allows me to do exactly what I want.
